erpnext/assets/doctype/asset_maintenance_log/asset_maintenance_log.py

```diff
@@ -75,6 +75,15 @@ def update_maintenance_task(self):
 		asset_maintenance_doc.save()
 
 
+def update_asset_maintenance_log_status():
+	maintenance_log = frappe.get_all(
```

Comment:
Instead of `get_all` and `for loop`, you can update all the relevant records in a single query using `qb.update`.